webapi/src/main/scala/org/knora/webapi/slice/admin/domain/service/KnoraUserService.scala

```diff
@@ -245,7 +247,30 @@ object KnoraUserService {
   )
 
   object Errors {
-    final case class UserServiceError(message: String)
+    sealed trait UserServiceError {
+      def message: String
+    }
+    private def msg(userIri: UserIri, reason: String, value: StringValue) =
+      s"User ${userIri.value} is $reason ${value.value}."
+
+    final case class NotGroupMember(userIri: UserIri, groupIri: GroupIri) extends UserServiceError {
+      override def message: String = msg(userIri, "is not member of group", groupIri)
+    }
```

Comment:
IMO `private def msg` is not necessary. I would merge it with below message, but if you want to keep it, remove `is` from the string, because it's doubled by passed `reason`.

```suggestion
    final case class NotGroupMember(userIri: UserIri, groupIri: GroupIri) extends UserServiceError {
      override def message: String = s"User ${userIri.value} is not member of group ${groupIri.value}."
    }
```